tools/kubernetes/migration.py

```diff
@@ -0,0 +1,153 @@
+def get_version_name(pipeline_name, version_display_name, index, add_prefix):
+    if add_prefix:
+       original_version_name = f"{pipeline_name}-{version_display_name}"
+    else:
+        original_version_name = f"{version_display_name}"    
+    
+    # Clean and convert the pipeline name to be k8s-compatible
+    formatted_name = utils.maybe_rename_for_k8s(original_version_name)
+
+    if len(formatted_name) > MAX_NAME_LENGTH:
+        suffix = f"-{index}"
+        max_name_len = MAX_NAME_LENGTH - len(suffix)
+        trimmed_name = original_version_name[:max_name_len]
+        formatted_name = utils.maybe_rename_for_k8s(trimmed_name + suffix)
+
+    return formatted_name
+
+# Convert a pipeline and its versions into k8s format
+def convert_to_k8s_format(pipeline, pipeline_versions, add_prefix):
+    k8s_objects = []
+
+    # Use namespace from pipeline object if available; fallback to default
+    namespace = pipeline.get('namespace') or K8S_NAMESPACE
+
+    # Clean and convert the pipeline name to be k8s-compatible
+    pipeline_name = utils.maybe_rename_for_k8s(pipeline.get('display_name', 'unnamed'))
+    versions = pipeline_versions.get("pipeline_versions", [])
+    
+    # Create the Pipeline object
+    pipeline_obj = {
+        "apiVersion": K8S_PIPELINE_API_VERSION,
+        "kind": "Pipeline",
+        "metadata": {
+            "name": pipeline_name,
+            "namespace": namespace,
+        }
+    }
+    k8s_objects.append(pipeline_obj)
+
+     # Create a PipelineVersion object for each version
+    for i, version in enumerate(versions):
+        version_display_name = version.get("display_name", f"v{i}")
+        pipeline_version_name = get_version_name(pipeline_name, version_display_name, i, add_prefix)        
+        pipeline_spec = version.get("pipeline_spec", {})
+        
+        pipeline_version_obj = {
+            "apiVersion": K8S_PIPELINE_API_VERSION,
+            "kind": "PipelineVersion",
+            "metadata": {
+                "name": pipeline_version_name,
+                "namespace": namespace
+            },
+            "spec": {
+                "pipelineName": pipeline_name,
+                "pipelineSpec": pipeline_spec
+            }
+        }
+        k8s_objects.append(pipeline_version_obj)
+   
+    return pipeline_name, k8s_objects
+
+# Write all collected Kubernetes objects to a seperate YAML file for each pipeline and its versions
+def write_pipeline_yaml(pipeline_name, k8s_objects, output_dir):
+    output_dir = Path(output_dir)
+    output_dir.mkdir(parents=True, exist_ok=True)
+    output_path = output_dir / f"{pipeline_name}.yaml"
+    with output_path.open('w') as f:
+        yaml.dump_all(k8s_objects, f, sort_keys=False)
+    print(f"Wrote pipeline '{pipeline_name}' to {output_path}")
+
+def migrate():
+    args = parse_args()
+    headers = {"Authorization": f"Bearer {args.token}", "Content-Type": "application/json"}
```

Comment:
I think the authorization header should only be set if `args.token` is set to a value.